srcpkgs/libaom/template

```diff
@@ -1,18 +1,40 @@
 # Template file for 'libaom'
 pkgname=libaom
-version=1.0.0.1
-revision=2
+version=2.0.0
+revision=1
 create_wrksrc=yes
 build_style=cmake
-configure_args="-DENABLE_DOCS=OFF -DBUILD_SHARED_LIBS=ON -DENABLE_TESTS=OFF"
+configure_args="-DENABLE_DOCS=OFF -DBUILD_SHARED_LIBS=ON"
 hostmakedepends="yasm doxygen perl python"
 short_desc="AV1 Codec library implementation"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="BSD-2-Clause"
 homepage="https://aomedia.org/"
 changelog="https://aomedia.googlesource.com/aom/+/master/CHANGELOG"
-distfiles="https://aomedia.googlesource.com/aom/+archive/v${version%.*}-errata${version##*.}.tar.gz"
-checksum=@0f1e812ca6d3b3809bf3f7a12a85b549c96d20bfa72fa606a682541bbc3eaeab
+distfiles="https://aomedia.googlesource.com/aom/+archive/v${version}.tar.gz"
+checksum=@4e8a74ed7e05d5534ab0b72735fa670ce1a1e757bc4a777f07fdabf8629fced1
+
+# aom segfaults with default musl stack size
+LDFLAGS="-Wl,-z,stack-size=2097152"
+
+case "$XBPS_TARGET_MACHINE" in
+	armv[67]l*) configure_args+=" -DENABLE_NEON=0" ;;
+esac
```

Comment:
@pullmoll should I include `arm5` here?

Reply:
you could just use `arm*` but i'd first check if it's necessary at all, it could be doing runtime checking (in which case we'd only disable it for armv5 and 6, and not 7)

Reply:
It broke the build for `armv7l` iirc, but I can test it again.

Reply:
Lots of similar errors:

```
[ 18%] Building C object CMakeFiles/aom_scale.dir/aom_scale/generic/yv12config.c.o
In file included from ../av1/common/arm/av1_txfm_neon.c:12:
../av1/common/arm/av1_txfm_neon.c: In function 'av1_round_shift_array_neon':
/usr/lib/gcc/armv7l-linux-musleabihf/9.3.0/include/arm_neon.h:6733:1: error: inlining failed in call to always_inline 'vdupq_n_s32': target specific option mismatch
 6733 | vdupq_n_s32 (int32_t __a)
      | ^~~~~~~~~~~
../av1/common/arm/av1_txfm_neon.c:23:37: note: called from here
   23 |   const int32x4_t dup_bits_n_32x4 = vdupq_n_s32((int32_t)(-bit));
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../av1/common/arm/av1_txfm_neon.c:12:
/usr/lib/gcc/armv7l-linux-musleabihf/9.3.0/include/arm_neon.h:10947:1: error: inlining failed in call to always_inline 'vst1q_s32': target specific option mismatch
10947 | vst1q_s32 (int32_t * __a, int32x4_t __b)
      | ^~~~~~~~~
../av1/common/arm/av1_txfm_neon.c:27:5: note: called from here
   27 |     vst1q_s32(arr, tmp_q_s32);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../av1/common/arm/av1_txfm_neon.c:12:
/usr/lib/gcc/armv7l-linux-musleabihf/9.3.0/include/arm_neon.h:4035:1: error: inlining failed in call to always_inline 'vrshlq_s32': target specific option mismatch
 4035 | vrshlq_s32 (int32x4_t __a, int32x4_t __b)
      | ^~~~~~~~~~
../av1/common/arm/av1_txfm_neon.c:26:17: note: called from here
   26 |     tmp_q_s32 = vrshlq_s32(tmp_q_s32, dup_bits_n_32x4);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../av1/common/arm/av1_txfm_neon.c:12:
/usr/lib/gcc/armv7l-linux-musleabihf/9.3.0/include/arm_neon.h:10369:1: error: inlining failed in call to always_inline 'vld1q_s32': target specific option mismatch
10369 | vld1q_s32 (const int32_t * __a)
      | ^~~~~~~~~
../av1/common/arm/av1_txfm_neon.c:25:27: note: called from here
   25 |     int32x4_t tmp_q_s32 = vld1q_s32(arr);
      |                           ^~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/aom_av1_common_neon_intrinsics.dir/build.make:83: CMakeFiles/aom_av1_common_neon_intrinsics.dir/av1/common/arm/av1_txfm_neon.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:393: CMakeFiles/aom_av1_common_neon_intrinsics.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 18%] Built target aom_decoder_app_util
```

Their armv7l toolchain has this, which implies lack of runtime detection.

```
set(AOM_NEON_INTRIN_FLAG "-mfpu=neon")

# No runtime cpu detect for armv7-linux-gcc.
set(CONFIG_RUNTIME_CPU_DETECT 0 CACHE STRING "")
```

Reply:
ok then, `arm*` it is

Reply:
Do I need anything for ppc? Or is it fine already?

Reply:
It adds the same thing for ppc:

```
set(CONFIG_RUNTIME_CPU_DETECT 0 CACHE STRING "")
```

So I guess we might need to disable simd (?).

Reply:
you can add `-DENABLE_VSX=0` for all `ppc*` except for `ppc64le*`, though chances are it's not necessary